tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,300 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file tizen-api.h
+ * @date 07 March 2019
+ * @brief Tizen NNStreamer/Pipeline(main) C-API Header.
+ *        This allows to construct and control NNStreamer pipelines.
+ * @see	https://github.com/nnsuite/nnstreamer
+ * @author MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#ifndef __TIZEN_NNSTREAMER_API_MAIN_H__
+#define __TIZEN_NNSTREAMER_API_MAIN_H__
+
+#include <nnstreamer/tensor_typedef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+/**
+ * @addtogroup CAPI_ML_NNSTREAMER_PIPELINE_MODULE
+ * @{
+ */
+
+/**
+ * @brief A handle of an NNStreamer pipeline.
+ * @since_tizen 5.5
+ */
+typedef struct nns_pipeline *nns_pipeline_h;
+
+/**
+ * @brief A handle of a "sink node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_sink *nns_sink_h;
+
+/**
+ * @brief A handle of a "src node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_src *nns_src_h;
+
+/**
+ * @brief A handle of a "switch" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_switch *nns_switch_h;
+
+/**
+ * @brief A handle of a "valve node" of an NNStreamer pipeline
+ * @since_tizen 5.5
+ */
+typedef struct nns_valve *nns_valve_h;
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer Pipelines.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_ERROR_NONE				= TIZEN_ERROR_NONE, /**< Success! */
+  NNS_ERROR_INVALID_PARAMETER			= TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  NNS_ERROR_NOT_SUPPORTED			= TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+} nns_error_e;
+
+/**
+ * @brief Enumeration for buffer deallocation policies.
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI)
+ */
+typedef enum {
+  NNS_BUF_FREE_BY_NNSTREAMER			= 0, /**< Default. Application should not deallocate this buffer. NNStreamer will deallocate when the buffer is no more needed */
+  NNS_BUF_NO_NOT_FREE				= 1, /**< This buffer is not to be freed. */
+} nns_buf_policy;
+
+/**
+ * @brief Enumeration for nns pipeline state
+ * @detail Refer to https://gstreamer.freedesktop.org/documentation/design/states.html
+ * @since_tizen 5.5
+ * @todo The list is to be filled! (NYI) Fill with GST's state values.
+ * @todo Check the consistency against MMFW APIs.
+ */
+typedef enum {
+  NNS_PIPELINE_UNKNOWN				= 0, /**< Unknown state. Maybe not constructed? */
```

Comment:
Do we need UNKNOWN state? 
What will be use of this state?

Reply:
Also, add in detail that "State of a pipeline is the state set for the pipeline. It does not imply that all the elements have reached this state. Some elements might have the current state of the pipeline as set in pending_state."
This description might fit better for nns_pipeline_getstate()

Reply:
> Do we need UNKNOWN state?

It may represent a state where a corresponding Gstreamer pipeline does not exist while the handle itself is not yet destroyed. With multithreaded environment, we may exprience such states.

> The description

I'll update it.